openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/ballerinatypegenerators/AllOfRecordTypeGenerator.java

```diff
@@ -103,23 +103,37 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws BallerinaOpenApiEx
                     typeName);
             return referencedTypeGenerator.generateTypeDescriptorNode();
         } else {
-            List<Node> recordFieldList = generateAllOfRecordFields(allOfSchemas);
-            addAdditionalSchemas(schema);
-            restDescriptorNode =
-                    restSchemas.size() > 1 ? getRestDescriptorNodeForAllOf(restSchemas) : restDescriptorNode;
-
-            NodeList<Node> fieldNodes = AbstractNodeFactory.createNodeList(recordFieldList);
-            return NodeFactory.createRecordTypeDescriptorNode(createToken(RECORD_KEYWORD),
-                    recordMetadata.isOpenRecord() ? createToken(OPEN_BRACE_TOKEN) : createToken(OPEN_BRACE_PIPE_TOKEN),
-                    fieldNodes, restDescriptorNode,
-                    recordMetadata.isOpenRecord() ? createToken(CLOSE_BRACE_TOKEN) :
-                            createToken(CLOSE_BRACE_PIPE_TOKEN));
+            ImmutablePair<List<Node>, List<Schema<?>>> recordFlist = generateAllOfRecordFields(allOfSchemas);
+            List<Node> recordFieldList = recordFlist.getLeft();
+            List<Schema<?>> validSchemas = recordFlist.getRight();
+            if (validSchemas.size() == 1) {
+                TypeGenerator typeGenerator = getTypeGenerator(validSchemas.get(0), typeName, null);
+                return typeGenerator.generateTypeDescriptorNode();
+            } else if (validSchemas.isEmpty()) {
+                AnyDataTypeGenerator anyDataTypeGenerator = new AnyDataTypeGenerator(schema, typeName);
+                return anyDataTypeGenerator.generateTypeDescriptorNode();
```

Comment:
Shall we change the order of these cases to be incremental?

Reply:
Updated in fc2dde1